### PR TITLE
feat(DDM): add ddm flavors data source

### PR DESCRIPTION
**Pull request by houpeng80**
**What this PR does / why we need it**:
  add ddm flavors data source
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
  add ddm flavors data source 
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc  TEST='./huaweicloud/services/acceptance/ddm/' TESTARGS='-run TestAccDatasourceDdmFlavors_basic' 
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/ddm/ -v -run TestAccDatasourceDdmFlavors_basic -timeout 360m -parallel 4 
=== RUN   TestAccDatasourceDdmFlavors_basic 
=== PAUSE TestAccDatasourceDdmFlavors_basic
=== CONT  TestAccDatasourceDdmFlavors_basic
--- PASS: TestAccDatasourceDdmFlavors_basic (9.05s) 
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/ddm       9.095s
```

### CODE REVIEW

File: docs/data-sources/ddm_flavors.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String) Specifies the region in which to query the data source.
+  If omitted, the provider-level region will be used.
+
+* `engine_id` - (Required, String) Specifies the ID of a engine.
+
+* `group_type` - (Optional, String) Specifies the compute resource architecture type. The options are **X86** and **ARM**.
```
Comment (ShiChangkuo) [inline docs/data-sources/ddm_flavors.md]: this option should be **cpu_arch**

File: docs/data-sources/ddm_flavors.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+* `type_code` - (Optional, String) Specifies the resource type code.
+
+* `code` - (Optional, String) Specifies the VM flavor types recorded in DDM.
+
+* `iaas_code` - (Optional, String) Specifies the VM flavor types recorded by the IaaS layer.
```
Comment (ShiChangkuo) [inline docs/data-sources/ddm_flavors.md]: this option is meaningless, please delete it

File: docs/data-sources/ddm_flavors.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+The following arguments are supported:
+
+* `region` - (Optional, String) Specifies the region in which to query the data source.
+  If omitted, the provider-level region will be used.
+
+* `engine_id` - (Required, String) Specifies the ID of a engine.
+
+* `group_type` - (Optional, String) Specifies the compute resource architecture type. The options are **X86** and **ARM**.
+
+* `type_code` - (Optional, String) Specifies the resource type code.
```
Comment (ShiChangkuo) [inline docs/data-sources/ddm_flavors.md]: it seems that the value is fixed to "hws.resource.type.ddm", I think you can delete it.

File: huaweicloud/services/ddm/data_source_huaweicloud_ddm_flavors.go [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+			"code": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `Indicates the VM flavor types recorded in DDM.`,
+			},
+			"iaas_code": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `Indicates the VM flavor types recorded by the IaaS layer.`,
+			},
```
Comment (ShiChangkuo) [inline huaweicloud/services/ddm/data_source_huaweicloud_ddm_flavors.go:118]: "vcpus" and "memory" should be added

File: docs/data-sources/ddm_flavors.md (hunk 1 above)
Comment (houpeng80) [reply docs/data-sources/ddm_flavors.md]: done

File: docs/data-sources/ddm_flavors.md (hunk 2 above)
Comment (houpeng80) [reply docs/data-sources/ddm_flavors.md]: done

File: docs/data-sources/ddm_flavors.md (hunk 3 above)
Comment (houpeng80) [reply docs/data-sources/ddm_flavors.md]: done

File: huaweicloud/services/ddm/data_source_huaweicloud_ddm_flavors.go (hunk 4 above)
Comment (houpeng80) [reply huaweicloud/services/ddm/data_source_huaweicloud_ddm_flavors.go:118]: done

File: huaweicloud/services/ddm/data_source_huaweicloud_ddm_flavors.go [hunk 5]
```diff
@@ -0,0 +1,291 @@
...
+	var mErr *multierror.Error
+
+	// getDdmFlavors: Query the List of DDM flavors
+	var (
+		getDdmFlavorsHttpUrl = "v2/{project_id}/flavors"
+		getDdmFlavorsProduct = "ddm"
+	)
+	getDdmFlavorsClient, err := cfg.NewServiceClient(getDdmFlavorsProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating DdmFlavors Client: %s", err)
```
Comment (ShiChangkuo) [inline huaweicloud/services/ddm/data_source_huaweicloud_ddm_flavors.go]:
```suggestion
		return diag.Errorf("error creating DDM client: %s", err)
```

File: huaweicloud/services/ddm/data_source_huaweicloud_ddm_flavors.go [hunk 6]
```diff
@@ -0,0 +1,291 @@
...
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"engine_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `Specifies the ID of a engine.`,
```
Comment (ShiChangkuo) [inline huaweicloud/services/ddm/data_source_huaweicloud_ddm_flavors.go]:
```suggestion
				Description: `Specifies the ID of an engine.`,
```

File: huaweicloud/services/ddm/data_source_huaweicloud_ddm_flavors.go [hunk 7]
```diff
@@ -0,0 +1,291 @@
...
+	}
+	curJson := utils.PathSearch("computeFlavorGroups", parma.resp, make([]interface{}, 0))
+	curArray := curJson.([]interface{})
+
+	var x86Flavors []interface{}
+	var armFlavors []interface{}
+	var offset, limit, x86Total, armTotal float64
+
+	for _, v := range curArray {
+		flavorCPUArch := utils.PathSearch("groupType", v, nil)
```
Comment (ShiChangkuo) [inline huaweicloud/services/ddm/data_source_huaweicloud_ddm_flavors.go]:
```suggestion
		flavorCPUArch := utils.PathSearch("groupType", v, “”)
```
should the  default value be ""?

**Pull request merged**